src/Foundation/NSExpression.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace XamCore.Foundation {
+
+	public partial class NSExpression {
+		bool hasFunction = false;
```

Comment:
The problem with managed state is that it might not be set if the instance is surfaced from native code, i.e. when using the `.ctor(IntPtr)`. In that case it would mean the `Block` would be `null` when it should not be.

From your commit message I assume that calling `Block` crash unless there's one ?